src/librustc_expand/expand.rs

```diff
@@ -363,7 +363,15 @@ impl<'a, 'b> MacroExpander<'a, 'b> {
                 krate.attrs = vec![];
                 krate.module = ast::Mod { inner: orig_mod_span, items: vec![], inline: true };
             }
-            _ => unreachable!(),
+            Some(ast::Item { span, kind, .. }) => {
+                self.cx.span_fatal(
```

Comment:
Seems like a fatal error would not be necessary here? Could do what the `None =>` branch above does and then use `cx.struct_span_err(...).emit()`.

Reply:
Oh true! We'd read through it, but somehow hadn't absorbed the fact that return is erroring out without a fatal error.

Thanks!